### PR TITLE
DOC: Update jnp.mean doc

### PR DESCRIPTION
**Pull request by rajasekharporeddy**
Updates the doc of `jnp.mean` and also fixes a typo in `jnp.sum` doc.
 
Part of #21461.

### CODE REVIEW

File: jax/_src/numpy/reductions.py [hunk 1]
```diff
@@ -676,10 +676,65 @@ def _axis_size(a: ArrayLike, axis: int | Sequence[int]):
...
+      If None, mean is computed along all the axes.
+    dtype: The type of the output array. Default=None.
+    keepdims: bool, default=False. If true, reduced axes are left in the result
+      with size 1.
+    where: int or array, default=None. The elements to be used in the mean. Array
+      should be broadcast compatible to the input.
+    out: Unused by JAX
+
+  Returns:
+    An array of the sum along the given axis.
```
Comment (jakevdp) [inline jax/_src/numpy/reductions.py:699]:
```suggestion
    An array of the mean along the given axis.
```

File: jax/_src/numpy/reductions.py [hunk 2]
```diff
@@ -676,10 +676,65 @@ def _axis_size(a: ArrayLike, axis: int | Sequence[int]):
...
+    Array([[2.5],
+           [4. ],
+           [5. ]], dtype=float32)
+
+    To use only specific elements of ``x`` to compute the mean, you can use
+    ``where``.
+
+    >>> where = jnp.array([[1, 0, 1, 0],
+    ...                    [0, 1, 0, 1],
+    ...                    [1, 1, 0, 1]])
```
Comment (jakevdp) [inline jax/_src/numpy/reductions.py:732]: Add `dtype=bool` for clarity.

**Review by jakevdp — CHANGES_REQUESTED** [review]

File: jax/_src/numpy/reductions.py [hunk 3]
```diff
@@ -676,10 +676,65 @@ def _axis_size(a: ArrayLike, axis: int | Sequence[int]):
...
 def mean(a: ArrayLike, axis: Axis = None, dtype: DTypeLike | None = None,
          out: None = None, keepdims: bool = False, *,
          where: ArrayLike | None = None) -> Array:
+  r"""Return the mean of array elements along a given axis.
+
+  JAX implementation of :func:`numpy.mean`.
+
+  Args:
+    a: Input array.
+    axis: int or array, default=None. Axis along which the mean to be computed.
```
[the remark this answers is not in the document]
Comment (rajasekharporeddy) [reply jax/_src/numpy/reductions.py]: Done

File: jax/_src/numpy/reductions.py [hunk 4]
```diff
@@ -676,10 +676,65 @@ def _axis_size(a: ArrayLike, axis: int | Sequence[int]):
...
+      If None, mean is computed along all the axes.
+    dtype: The type of the output array. Default=None.
+    keepdims: bool, default=False. If true, reduced axes are left in the result
+      with size 1.
+    where: int or array, default=None. The elements to be used in the mean. Array
```
[the remark this answers is not in the document]
Comment (rajasekharporeddy) [reply jax/_src/numpy/reductions.py]: Done

File: jax/_src/numpy/reductions.py (hunk 1 above)
Comment (rajasekharporeddy) [reply jax/_src/numpy/reductions.py]: Done

File: jax/_src/numpy/reductions.py (hunk 2 above)
Comment (rajasekharporeddy) [reply jax/_src/numpy/reductions.py]: Done

**Comment by rajasekharporeddy** [comment]
Implemented the review changes and squashed into single commit.

**Review by dfm — APPROVED** [review]
Looks good - thanks!

**Pull request merged**